notebooks/07 - Crossfilter.ipynb

```diff
@@ -0,0 +1,407 @@
+    "    display(unfolded_map)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d854e55b-83ed-4a58-a861-9a31f18dfe08",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "df = pd.read_csv('./data/earthquakes.csv')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93c4b4ca-5842-44b5-a8c4-7edf96b64929",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "unfolded_map.add_dataset({\n",
+    "    'label': 'Earthquakes',\n",
+    "    'data': df\n",
+    "})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d37de93a-76c9-4da9-a3e1-8d903ae9e533",
+   "metadata": {},
+   "source": [
+    "## Syncing a map filter with a Plotly histogram"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7052b802-abb2-491d-a93a-120a197612cb",
+   "metadata": {},
+   "source": [
+    "We can use Plotly to draw a histogram showing the distribution of the numbers of earthquakes by their magnitude:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9c1c0fd7-4740-48bf-867e-07984c45afbb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import plotly.graph_objects as go\n",
+    "\n",
+    "hist_fig = go.FigureWidget([go.Histogram(x = df['Magnitude'], nbinsx = 50)])\n",
+    "hist = hist_fig.data[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c5a4f964-75b1-4d9c-adb3-8574bd1ebd48",
+   "metadata": {},
+   "source": [
+    "Let's now add event handlers. This function will be called when the selection in the histogram is changed:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1a7784fe-3a59-49ae-9419-76a5d91459e4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def on_histogram_selection_change(trace, points, state):\n",
+    "    # Update the Magnitude filter in the map\n",
+    "    unfolded_map.set_filter({\n",
+    "        'id': 'magnitude_filter',\n",
+    "        'field': 'Magnitude',\n",
+    "        'value': [min(points.xs), max(points.xs)]\n",
+    "    })\n",
+    "hist_fig.data[0].on_selection(on_histogram_selection_change)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "01f2c857-3dd4-476a-9afd-e2b637df64e0",
+   "metadata": {},
+   "source": [
+    "This one will be called when a filter in the map is changed:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "70925541-1928-43f7-b90a-b226bb9e3495",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def on_map_filter_change(info):\n",
+    "    # Update the selection in the histogram\n",
+    "    if 'Magnitude' in info['name'] and info['prop'] == 'value':\n",
+    "        v = info['value']\n",
+    "        condition = df['Magnitude'].between(v[0], v[1])\n",
+    "        indices = tuple(df.index[condition])\n",
+    "        hist.selectedpoints = indices\n",
+    "\n",
+    "unfolded_map.set_map_event_handlers({\n",
```

Comment:
Maybe add a link to the docs in the description above?

Reply:
ok will add